src/connectors/configure/connectConfigure.ts

```diff
@@ -132,13 +132,23 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
       },
 
       getRenderState(renderState, renderOptions) {
+        const widgetRenderState = this.getWidgetRenderState(renderOptions);
         return {
           ...renderState,
-          // Even if there are multiple configure widgets,
-          // the last configure widget will override the ones before.
-          // If we want to merge widgetRenderState of multiple configure widgets,
-          // we should modify this part.
-          configure: this.getWidgetRenderState(renderOptions),
+          configure: {
+            ...widgetRenderState,
+            widgetParams: {
+              ...widgetRenderState.widgetParams,
+              searchParameters: mergeSearchParameters(
+                new algoliasearchHelper.SearchParameters(
+                  renderState.configure?.widgetParams.searchParameters
```

Comment:
Not sure if it's needed but we could add a fallback (`|| {}`) in case configure is `undefined`?

Reply:
it's not needed, since undefined is a valid input for search parameters :)